app/Http/Controllers/DeviceController.php

```diff
@@ -126,4 +140,68 @@ private function getDeviceGraphs(Device $device)
         $os_group = Config::getOsSetting($device->os, 'group');
         return Config::get("os.$os_group.over", Config::get('os.default.over'));
     }
+
+    private function deviceLinkMenu(Device $device)
+    {
+        $can_admin_device = $device->canAccess(Auth::user());
```

Comment:
This doesn't check admin access. Also, we should probably start using policy or gates

Reply:
Actually it does, but you are right, should use the gate.

Reply:
The Permissions::canAccessDevice() does not check admin access ;)